query/physicalplan/sampler.go

```diff
@@ -174,6 +174,10 @@ func (s *ReservoirSampler) sliceReservoir() {
 
 // sample implements the reservoir sampling algorithm found https://en.wikipedia.org/wiki/Reservoir_sampling.
 func (s *ReservoirSampler) sample(r arrow.Record, ref *referencedRecord) {
+	// The size can be 0 and in that case we don't want to sample.
+	if s.size == 0 {
+		return
+	}
```

Comment:
@thorfour, this sometimes panics all of a sudden. 
Specifically line 189/193 with the `rand.Intn(int(s.size)`. 
I'm not sure why we're sometimes running into this now. But this guard at least stops the panic.